File: problems/409-longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        count = {}
        for l in s:
            if l in count:
                count[l] += 1
            else:
                 count[l] = 1

        result = 0
        flag = False
        for value in count.values():
            if value % 2 == 0:
                result += value
            else:
                result += value - 1
                flag = True
        
        if flag == True:
            result += 1
        
        return result

File: problems/409-longest-palindrome/longest-palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
const longestPalindrome = (s) => {
    const hashCount = {};
    for (let i = 0; i < s.length; i++) {
        hashCount[s[i]] = (hashCount[s[i]] || 0) + 1;
    }

    let result = 0;
    let flag = false;
    for (let value of Object.values(hashCount)) {
        if (value % 2 === 0) {
            result += value;
        } else {
            result += value - 1;
            flag = true;
        }
    }
    if (flag) result++;
    return result;
};